src/ds.erl

```diff
@@ -209,6 +210,17 @@ map(Obj,Fun) when is_list(Obj) andalso is_function(Fun, 2) ->
 map(Obj,Fun) when (is_function(Fun, 1) orelse is_function(Fun, 2)) ->
     as_list(Obj, fun(List) -> map(List, Fun) end).
 
+-spec foreach(object(), fun()) -> object().
```

Comment:
Thanks David! The one thing I'm noticing here is that the return value here is listed as `object()`, which the return value for `foreach` should just be `ok`.

Reply:
The spec return value was fixed